pkg/client/block/client.go

```diff
@@ -36,42 +39,147 @@ func NewBlockClient(
 	eventsReplayClient client.EventsReplayClient[client.Block]
+
+	// latestBlockReplayObs is a replay observable that combines blocks observed by
+	// the block query client & the events replay client. It is the "canonical"
+	// source of block notifications for blockClient.
+	latestBlockReplayObs observable.ReplayObservable[client.Block]
+
+	// close is a function that cancels the context of the blockClient.
+	close context.CancelFunc
 }
 
 // CommittedBlocksSequence returns a replay observable of new block events.
-func (b *blockClient) CommittedBlocksSequence(ctx context.Context) client.BlockReplayObservable {
+func (b *blockReplayClient) CommittedBlocksSequence(ctx context.Context) client.BlockReplayObservable {
 	return b.eventsReplayClient.EventsSequence(ctx)
 }
 
-// LatestsNBlocks returns the last n blocks observed by the BockClient.
-func (b *blockClient) LastNBlocks(ctx context.Context, n int) []client.Block {
-	return b.eventsReplayClient.LastNEvents(ctx, n)
+// LastBlock returns the last blocks observed by the BlockClient.
+func (b *blockReplayClient) LastBlock(ctx context.Context) (block client.Block) {
+	// ReplayObservable#Last() is guaranteed to return at least one element.
+	return b.latestBlockReplayObs.Last(ctx, 1)[0]
 }
 
 // Close closes the underlying websocket connection for the EventsQueryClient
 // and closes all downstream connections.
-func (b *blockClient) Close() {
+func (b *blockReplayClient) Close() {
 	b.eventsReplayClient.Close()
+	//close(b.latestBlockPublishCh)
+	b.close()
+}
+
+// forEachBlockEvent asynchronously observes block event notifications from the
+// EventsReplayClient's EventsSequence observable & publishes each to latestBlockPublishCh.
+func (b *blockReplayClient) forEachBlockEvent(
+	ctx context.Context,
+	latestBlockPublishCh chan<- client.Block,
+) {
+	channel.ForEach(ctx, b.eventsReplayClient.EventsSequence(ctx),
+		func(ctx context.Context, block client.Block) {
+			latestBlockPublishCh <- block
+		},
+	)
+}
+
+// getInitialBlock fetches the latest committed on-chain block at the time the
+// client starts up, while concurrently waiting for the next block event,
+// publishing whichever occurs first to latestBlockPublishCh.
+// This is necessary to ensure that the most recent block is available to the
+// blockClient when it is first created.
+func (b *blockReplayClient) getInitialBlock(
+	ctx context.Context,
+	latestBlockPublishCh chan<- client.Block,
+) error {
+	blockQueryResultCh := make(chan client.Block)
+
+	// Query the latest block asynchronously.
+	queryErrCh := b.queryLatestBlock(ctx, blockQueryResultCh)
+
+	// Wait for either the latest block query response, error, or the first block
+	// event to arrive & use whichever occurs first or return an error.
+	//
+	// NB: #latestBlockReplayObs is a proxy for the events sequence observable
+	// because it is guaranteed to be notified on block events in #goForEachBLockEvent().
+	var initialBlock client.Block
```

Comment:
1. goForEachBLockEvent no longer ecists
2. This comment is adding confusion rather than clarity for me

```suggestion
	var initialBlock client.Block
```